src/snowflake/snowpark/_internal/telemetry.py

```diff
@@ -168,6 +168,10 @@ def wrap(*args, **kwargs):
         ]._session.sql_simplifier_enabled
         try:
             api_calls[0][TelemetryField.QUERY_PLAN_HEIGHT.value] = plan.plan_height
+            # The uuid for df._select_statement can be different from df._plan. We always
```

Comment:
I am little bit confused of the relationship between df._select_statement and _plan here. what is the purpose of mentioning df._select_statement here?

Reply:
I should be more clear here. `plan = args[0]._select_statement or args[0]._plan` therefore we cannot use `plan.uuid` value for telemetry. I will update the comment with this explanation.